src/Ucommerce.Sitecore/Content/SitecoreImageService.cs

```diff
@@ -42,11 +43,39 @@ public virtual Ucommerce.Content.Content GetImage(string contentId)
 				return content;
 			}
 
-			content.Name = item.Name;
-			content.Url = GetMediaUrl(item);
-		    content.Icon = item.Appearance.Icon;
+			return MapItemToContent(item);
+		}
+
+		protected virtual Ucommerce.Content.Content MapItemToContent(Item item)
+		{
+			return new Ucommerce.Content.Content()
+			{
+				Name = item.Name,
+				Icon = item.Appearance.Icon,
+				Url = GetMediaUrl(item),
+				Id = item.ID.ToString(),
+				// NodeType = item.Template.Key == "media folder" ? "Folder" : "Image"
```

Comment:
Has this been left in place intentionally as a reminder for some sort of next step of the fix/refactoring?
If not, please remove it.

Reply:
Was about to write the same comment 😂

Reply:
Yeah, I thought I made this a draft (no matching story in the review column on CH at the time.)